reana_job_controller/compute4punch_job_manager.py

```diff
@@ -0,0 +1,366 @@
+        :param kwargs: Additional parameters needed to fetch logs.
+            In the case of Slurm, the ``workspace`` parameter is needed.
+        :return: String containing the job logs.
+        """
+        if "workspace" not in kwargs:
+            raise ValueError("Missing 'workspace' parameter")
+        workspace = kwargs["workspace"]
+
+        job_log = ""
+
+        try:
+            for log in ("out", "err"):
+                filepath = f"logs/{backend_job_id}.{log}"
+                with open_file(workspace, filepath) as f:
+                    job_log += f.read()
+            return job_log
+        except FileNotFoundError as e:
+            msg = f"Job logs of {backend_job_id} were not found. {e}"
+            logging.error(msg, exc_info=True)
+            return msg
+
+    @classmethod
+    def get_outputs(cls, c4p_connection, src, dest) -> None:
+        """
+        Transfer job outputs from Compute4PUNCH to local REANA workspace.
+        """
+        sftp_client = c4p_connection.ssh_client.open_sftp()
+        sftp_client.chdir(src)
+        try:
+            cls._download_output_directory(sftp_client, src, dest)
+        finally:
+            sftp_client.close()
+
+    def stop(self, backend_job_id: str) -> None:
+        """
+        Stop job execution.
+
+        :param backend_job_id: The backend job id
+        :type backend_job_id: str
+        """
+        try:
+            self.c4p_connection.exec_command(f"condor_rm {backend_job_id}")
+        except Exception as ex:
+            logging.error(ex, exc_info=True)
+
+    @property
+    def c4p_home_path(self) -> str:
+        """
+        Determine and return the Compute4PUNCH home directory on Compute4PUNCH
+        """
+        if not self.C4P_HOME_PATH:
+            # Since the JobMonitor entirely rely on class variables to get corresponding
+            # paths on Compute4PUNCH, the class variable C4P_HOME_PATH needs to be
+            # modified here.
+            Compute4PUNCHJobManager.C4P_HOME_PATH = self.c4p_connection.exec_command(
+                "pwd"
+            ).strip()
+        return self.C4P_HOME_PATH
+
+    @property
+    def c4p_abs_workspace_path(self) -> str:
+        """
+        Determine and return the absolute Compute4PUNCH workspace path
+        """
+        if not self.C4P_WORKSPACE_PATH:
+            # Since the JobMonitor entirely rely on class variables to get corresponding
+            # paths on Compute4PUNCH, the class variable C4P_WORKSPACE_PATH needs to be
+            # modified here.
+            Compute4PUNCHJobManager.C4P_WORKSPACE_PATH = os.path.join(
+                self.c4p_home_path, self.c4p_rel_workspace_path
+            )
+        return self.C4P_WORKSPACE_PATH
+
+    @property
+    def c4p_rel_workspace_path(self) -> str:
+        """
+        Determine and return the relative Compute4PUNCH workspace path
+        """
+        return os.path.join("reana/workflows", self.workflow_uuid)
```

Comment:
It represents the path relative to the users home directory, where the REANA workflow space is located on the Compute4PUNCH login node. I agree that it would be nice to make this configurable by an environment variable. Fixed in https://github.com/reanahub/reana-job-controller/pull/430/commits/97c63c9e42d5df43d5b542d62c60e33610141559